rpcclient/infrastructure.go

```diff
@@ -1338,6 +1343,13 @@ func newHTTPClient(config *ConnConfig) (*http.Client, error) {
 		},
 	}
 
+	// Configure the client to use a Unix socket if one is configured.
+	if config.UnixSocketPath != "" {
```

Comment:
Can't we do this in a bit of a cleaner way? I think we don't need another field and can just parse the host into a `net.Address` (example: https://github.com/lightningnetwork/lnd/blob/cd566eb0973240667d37823aaee2b5a61a012173/lncfg/address.go#L209). Then we can always set a dialer on the client:

```
	Transport: &http.Transport{
		DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
			return net.Dial(parsedAddr.Network(), parsedAddr.String())
		},
	},
```

Reply:
Thanks, this is a much cleaner implementation, I'll work on it

Reply:
Would it be more advantageous to duplicate a portion of that code instead of importing the function?
 It appears that the function performs additional actions and requires a defaultPort and tcpResolver.

Reply:
Would it be preferable to duplicate the code instead of adding a new import for `"github.com/lightningnetwork/lnd/lncfg"`

Reply:
Yes, definitely. Otherwise it's easy to get circular package dependencies. Also, I don't think we need Tor/onion support, so that part isn't needed.